src/redis-cli.c

```diff
@@ -1266,6 +1267,10 @@ static int cliReadReply(int output_raw_strings) {
                 slot, config.hostip, config.hostport);
         config.cluster_reissue_command = 1;
         cliRefreshPrompt();
+    } else if (config.set_errcode && reply->type == REDIS_REPLY_ERROR) {
```

Comment:
This will cause an unintentional exit in REPL/interactive mode.

```suggestion
    } else if (!config.interactive && config.set_errcode &&
               reply->type == REDIS_REPLY_ERROR) {
```

Reply:
thanks @itamarhaber , I think te propose of -e flag means fail if a command execution returns error, however, considering about the case if user both provide -e and running in interactive mode, i would vote for user wants to exit interactive mode when an error happens, how do you think? or did i miss something here?

Reply:
i agree the user might be an idiot if he passed the `-e`, but i think it makes no sense (i suppose @itamarhaber suggestion)

Reply:
thanks @oranagra @itamarhaber , I will change the code for this part

Reply:
I specialize in idiocy, ergo my suggestion. QED.